data/data-pipeline/data_pipeline/etl/sources/census/etl_utils.py

```diff
@@ -69,3 +72,35 @@ def get_state_information(data_path: Path) -> pd.DataFrame:
     df["fips"] = df["fips"].astype(str).apply(lambda x: x.zfill(2))
 
     return df
+
+
+def check_census_data(census_data_path: Path, census_data_source: str) -> None:
+    """Checks if census data is present, and exits gradefully if it doesn't exist and if the user didn't request S3 downloading
+
+    Args:
+        census_data_path (str): Path for Census data
+        census_data_source (str): Source for the census data
+                                  Options:
+                                  - local: fetch census data from the local data directory
+                                  - aws: fetch census from AWS S3 J40 data repository
+
+    Returns:
+        None
+
+    """
+    CENSUS_DATA_S3_URL = settings.AWS_JUSTICE40_DATASOURCES_URL + "/census.zip"
+    DATA_PATH = settings.APP_ROOT / "data"
+
+    if not os.path.isfile(census_data_path / "geojson" / "us.json"):
+        if census_data_source == "aws":
+            logger.info("Fetching Census data from AWS S3")
+            unzip_file_from_url(
```

Comment:
Well, if the user has the census data locally, why download again? Bear in mind the function could be called more than one time in different points of the ETL.. So the function is checking if there's no census data locally, then check if the user wants to use "aws", otherwise throw an error -- but open to discussing.

Reply:
You could have a corrupted, incomplete, or out of date local dataset and want to use the version from AWS instead (or  simply want to confirm that you're using what production is using). If I didn't want to use my local dataset for some reason, I'd pass the "aws" flag and then be super confused about why it wasn't actually using aws. I wouldn't know that I'd need to delete my local dataset first before proceeding.

Reply:
So I think we should either change the logic to always let the user's choice take precedence, or update the code to output a warning telling the user that we're using the local version even though they gave the "aws" flag.

Reply:
> Bear in mind the function could be called more than one time in different points of the ETL
with this in mind maybe then I'd advocate for the latter, having an output log saying that regardless of the aws flag we are using the local version. I think it'd be pretty difficult to do the former option if we had to keep track that we'd already gotten it from AWS on the 2nd or 3rd time the command was run in the ETL pipeline.

Reply:
Upon further analysis, makes sense to download always if "aws" is being passed, as the score and combilefy functions will be run in different GH actions.